compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -975,7 +1054,8 @@ fn determine_capture_info(
     };
 
     if eq_capture_kind {
-        match (capture_info_a.expr_id, capture_info_b.expr_id) {
+        // FIXME: Should this account for both expr_id fields?
```

Comment:
I think it's fine to leave the responsibility of path size on the caller, but we should document that

Reply:
That said we probably want to prioritize path_expr=Some(..) over None